File: Engr-T.stark/containerWithMostWater.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1, curMax = 0, area;
        
        while(left < right)
        {
            area = min(height[left], height[right]) * (right - left);
            curMax = max(curMax, area);
            if(height[left] < height[right]) left++;
            else { right--; }
        }
        return curMax;
    }
};

File: Eff_Emm/container_with_most_water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left = 0
        right = len(height) -1
        result = 0
        start = 0
        
        while left < right:
            area = min(height[left], height[right]) *( right-left)
            result = max(result, area)
            
            if height[right] > height[left]:
                left += 1
                
            elif height[left] > height[right]:
                right -= 1
            else:
                right -=1
            
            
        return result
            